doc/development_guide/testing.rst

```diff
@@ -0,0 +1,131 @@
+We do recommend using the tox_ command though::
+
+    pytest pylint -k test_functional
+
+Writing functional tests
+------------------------
+
+These are residing under '/pylint/test/functional' and they are formed of multiple
+components. First, each Python file is considered to be a test case and it
+should be accompanied by a .txt file, having the same name, with the messages
+that are supposed to be emitted by the given test file.
+
+In the Python file, each line for which Pylint is supposed to emit a message
+has to be annotated with a comment in the form ``# [message_symbol]``, as in::
+
+    a, b, c = 1 # [unbalanced-tuple-unpacking]
+
+If multiple messages are expected on the same line, then this syntax can be used::
+
+    a, b, c = 1.test # [unbalanced-tuple-unpacking, no-member]
+
+You can also use ``# +n: [`` with n an integer if the above syntax would make the line too long or other reasons::
+
+    # +1: [empty-comment]
+    #
+
+If you need special control over Pylint's configuration, you can also create a .rc file, which
+can have sections of Pylint's configuration.
+
+During development, it's sometimes helpful to run all functional tests in your
+current environment in order to have faster feedback. Run from Pylint root directory with::
+
+    python tests/test_functional.py
+
+You can use all the options you would use for pytest, for example `-k "test_functional[len_checks]"`.
+Furthermore, if required the .txt file with expected messages can be regenerated based
+on the the current output by appending `--update-functional-output` to the command line::
+
+    python tests/test_functional.py --update-functional-output -k "test_functional[len_checks]"
+
+Writing unittest tests
+------------------------
+
+Most other tests reside in the '/pylint/test' directory. These unittests can be used to test
+almost all functionality within Pylint. A good step before writing any new unittests is to look
+at some tests that test a similar funcitionality. This can often help write new tests.
+
+If your new test requires any additional files you can put those in the
+`/pylint/test/regrtest_data` directory. This is the directory we use to store any data needed for
+the unittests.
+
+
+Writing functional tests for configurations
+-------------------------------------------
+
+To test the different way to configure Pylint there is also a small functional test framework
+for configuration files. These tests can be found in the '/pylint/test/config' directory.
+
+To create a new test create a new file with an unused name in the directory of that type
+of configuration file. Subsequently add a `.json` file with the same name which records
+what the configuration should be **compared to the standard configuration**.
+
+For example, if the configuration should add a warning to the list of disabled messages the
+`.json` file should include::
+
+    "functional_append": {
+        "disable": [["a-message-to-be-added"],]
+    }
+
```

Comment:
```suggestion
For example, if the configuration should add a warning to the list of disabled messages
and you changed the configuration for ``job`` to 10 instead of the default 1 the
`.json` file should include::

    "functional_append": {
        "disable": [["a-message-to-be-added"],]
    }
    "jobs": 10,

```

Reply:
👍